### DISCUSSION TITLE
Add a feature to drop first and last data points for alerts

### DISCUSSION THREAD

**Discussion by derN3rd** [grafana/grafana#40552] [Alerting]
cc #25234

Now with the new Grafana 8.0 alerting I find it hard to have my alerts working in a good way, as they go randomly off when they check a not completed timeframe.
Like seen in the linked issue above, the first and the last datapoint completely mess up the avg, first()/last() values and that makes it impossible for us to use them in out alertings.

Is there any workaround in general or for the mysql datasource? (e.g. like the [ClickHouse datasource](https://grafana.com/grafana/plugins/vertamedia-clickhouse-datasource/) does it with Extrapolation and _round time_)

Thanks in advance

**Comment by worker24h** [comment]
I have the same problem. I would like to add a Transformation Function to solve it

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
